pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -42,9 +61,125 @@ public long getInitialDelayInSeconds() {
     return _initialDelayInSeconds;
   }
 
+  /**
+   * Returns the status of the {@code started} flag. This flag will be set after calling {@link #start()}, and reset
+   * after calling {@link #stop()}.
+   */
+  public final boolean isStarted() {
+    return _started;
+  }
+
+  /**
+   * Returns the status of the {@code running} flag. This flag will be set during the task execution.
+   */
+  public final boolean isRunning() {
+    return _running;
+  }
+
+  /**
+   * {@inheritDoc}
+   * <p>
+   * This method sets {@code started} flag to true.
+   */
   @Override
-  public String getTaskName() {
-    return _taskName;
+  public final synchronized void start() {
```

Comment:
So the purpose of adding synchronized here is to avoid different periodic tasks running at the same time? Then is it the same as running multiple tasks in one thread?

Reply:
No, the purpose of this synchronized is to avoid calling start() and stop() at the same time. Basically another level of protection. (start() and stop() are called in scheduler, not in the task thread)